### PR TITLE
cloudflare lambda build and deploy pipeline

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
- adds Cloudflare Lambda ("worker") build and deploy pipeline, in parallel with the existing AWS Lambda pipeline 
- adds explanatory README (please read before reviewing) 

## WHY
- we plan to migrate the rematch service from AWS to Cloudflare 

## HOW
1. Add wrangler directory and environment - wrangler is the CLI that Cloudflare uses to manage and deploy serverless functions. 
2. In order to minimize the scope of the migration, and to avoid regressions, we continue to build the Cloudflare version within the AWS directory, and deploy the build artifact.

## Notes 
`Cloudflare Workers` is a more specialized service compared with `AWS Lambda`. The "API Gateway" is already built in / implicit, whereas one must specifically prefix it to an `AWS Lambda` instance. `Cloudflare` also assumes all `Worker` responses are HTTP responses. 

I'm happy to do a quick tour of Cloudflare UIs that are relevant to this PR - just ping me 👋 

## Rollout 
After deploying this PR, cutover from AWS to Cloudflare is simple and reversible: 
`heroku config:set REMATCH_LAMBDA_URL=<target>` 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Migrate-Rematch-to-Cloudflare-b3af455baca9480581d4b06e0b987a32

### What have you done to QA this feature?
- set up Postman environments for testing convenience
- retrieved an known-working AWS lambda payload from staging, and added it to the Postman requests
- confirmed that the responses to the payload were identical between AWS staging and Cloudflare staging

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  no - scripts only
Have you deployed to Staging? | n/a
Self-Review: Have you done an initial self-review of the code below on Github? |

### CODE REVIEW

File: lambdas/.nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+lts/iron
```
Comment (happythenewsad) [inline lambdas/.nvmrc:1]: Required for `wrangler init`, a one-time setup operation

File: lambdas/cloudflare_rematch/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (emilia-friedberg) [inline lambdas/cloudflare_rematch/tsconfig.json:1]: It feels like there's a lot of commented-out code in this file -- maybe we should just keep the link to whatever the reference is in a comment and delete what we aren't using?

**Review by emilia-friedberg — APPROVED** [review]
As long as it works it LGTM! Left a comment about comment clean up (haha) but no need to review again.

File: lambdas/rematching/deploy.sh [hunk 3]
```diff
@@ -1,5 +1,14 @@
 #!/bin/bash
 
+read -r -p "You are attempting to deploy rematch to AWS. Have you read the README and want to proceed? [y]" response
```
Comment (anathomical) [inline lambdas/rematching/deploy.sh:3]: I really like this solution for supporting our temporary support for both pieces of infrastructure.

File: lambdas/cloudflare_rematch/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Welcome to Cloudflare Workers! This is your first worker.
+ *
+ * - Run `wrangler dev src/index.ts` in your terminal to start a development server
+ * - Open a browser tab at http://localhost:8787/ to see your worker in action
+ * - Run `wrangler deploy src/index.ts --name my-worker` to deploy your worker
+ *
+ * Learn more at https://developers.cloudflare.com/workers/
+ */
```
Comment (anathomical) [inline lambdas/cloudflare_rematch/src/index.ts:9]: Echoing @emilia-friedberg 's comment on the `tsconfig.json` file: I'm not sure it's worth keeping the boiler-plate commented-out code for documentation purposes.

**Review by anathomical — APPROVED** [review]
I think it's probably worth stripping out a bunch of the boiler-plate commented out stuff and rely on documentation if anyone needs to change any config options.

Additionally, I went through the instructions in the `README.md` and was able to successfully deploy to Cloudflare staging.

**Pull request merged**